src/main/java/com/blazemeter/ciworkflow/CiPostProcess.java

```diff
@@ -83,16 +89,32 @@ public CiPostProcess(boolean isDownloadJtl, boolean isDownloadJunit, String jtlP
         this.logger = logger;
     }
 
+    public void setTest(String testId, String testType) {
+        this.testId = testId;
+        this.testType = testType;
+    }
+
     /**
      * Executes post-process after test was finished on server
      *
      * @return BuildResult
      */
     public BuildResult execute(Master master) {
         try {
-            JSONObject ciStatus = master.getCIStatus();
-            BuildResult result = validateCiStatus(ciStatus);
+            JSONObject ciStatus;
+            BuildResult result;
+
+            if (testType.equals(FUNCTIONAL_GUI_TEST)) {
+                Thread.sleep(15000);
```

Comment:
why you need 15 sec sleep here?

Reply:
For checking status for functional test, calling /api/v4/masters/{master_Id} API gives response with the summary object as null, So result showing wrong because response having a summary object with definedStatus value which is not returning in API response. When added timesleep of 15 sec API returns the summary object in response and then it shows the correct test result.

Reply:
IMO it's better to handle response with `null` status on this side and try to send several more requests to this API with interval 3-5 sec.
e.g.
```
startTime = currentTIme()
while (res =getFunctionalStatus) != null && waitTime < 60 (sec) {
  waitTIme = currentTime() - startTime;
  log.info("result is empty, waiting for 5 seconds")
  Thread.sleep(5000)
} 
```

It's much better than sleep default time, because we calculate this summary in workers and it theoretically can be done in more than 15 second